oximeter/db/src/native/block.rs

```diff
@@ -955,4 +1019,63 @@ mod tests {
             ValueArray::UInt64([data.as_slice(), data.as_slice()].concat())
         );
     }
+
+    #[test]
+    fn test_parse_enum_variant() {
+        assert_eq!(enum_variant("'Foo' = 1'").unwrap().1, (1, "Foo"),);
+        assert_eq!(enum_variant("\\'Foo\\' = 1'").unwrap().1, (1, "Foo"),);
+
+        enum_variant("'Foo'").unwrap_err();
+        enum_variant("'Foo' = ").unwrap_err();
+        enum_variant("'Foo' = x").unwrap_err();
+        enum_variant("\"Foo\" = 1").unwrap_err();
+    }
+
+    #[test]
+    fn test_parse_enum8() {
+        let parsed = enum8("Enum8('Foo' = 1, 'Bar' = 2)").unwrap().1;
+        let DataType::Enum8(map) = parsed else {
+            panic!("Expected DataType::Enum8, found {parsed:#?}");
+        };
+        assert_eq!(map.len(), 2);
+        assert_eq!(map.get(&1).unwrap(), "Foo");
+        assert_eq!(map.get(&2).unwrap(), "Bar");
+    }
+
+    #[test]
+    fn test_parse_array_enum8_with_escapes() {
+        const INPUT: &str = r#"Array(Enum8(\'Bool\' = 1, \'I64\' = 2))"#;
```

Comment:
I'm totally spitballing here, so feel free to ignore. I have no idea if this makes sense. 

Is it worth trying to see what happens if you try to parse with extra backslashes or escape sequences and see if we error out as expected?

Reply:
Done in 997682132